src/rust/src/x509.rs

```diff
@@ -19,6 +19,43 @@ lazy_static::lazy_static! {
     static ref CRL_REASON_OID: asn1::ObjectIdentifier<'static> = asn1::ObjectIdentifier::from_string("2.5.29.21").unwrap();
     static ref CRL_NUMBER_OID: asn1::ObjectIdentifier<'static> = asn1::ObjectIdentifier::from_string("2.5.29.20").unwrap();
     static ref DELTA_CRL_INDICATOR_OID: asn1::ObjectIdentifier<'static> = asn1::ObjectIdentifier::from_string("2.5.29.27").unwrap();
+    static ref SUBJECT_ALTERNATIVE_NAME_OID: asn1::ObjectIdentifier<'static> = asn1::ObjectIdentifier::from_string("2.5.29.17").unwrap();
+}
+
+struct UnvalidatedIA5String<'a>(&'a str);
+
+impl<'a> asn1::SimpleAsn1Readable<'a> for UnvalidatedIA5String<'a> {
+    const TAG: u8 = 0x16;
+    fn parse_data(data: &'a [u8]) -> asn1::ParseResult<Self> {
+        Ok(UnvalidatedIA5String(
+            std::str::from_utf8(data).map_err(|_| asn1::ParseError::InvalidValue)?,
+        ))
+    }
+}
+
+#[derive(asn1::Asn1Read)]
+enum GeneralName<'a> {
+    #[implicit(0)]
+    OtherName(AttributeTypeValue<'a>),
+    #[implicit(1)]
```

Comment:
nit: can you put a newline between each of the options? I think it'll make it more readable